src/main/kotlin/com/exactpro/th2/conn/amqp/impl/ConnServiceImpl.kt

```diff
@@ -33,19 +32,26 @@ class ConnServiceImpl(
 
     private lateinit var client: AmqpClient
 
+    private fun toMap(parameters: ConnParameters): Map<String, String> {
+        val environmentDetails = HashMap<String, String>()
+        environmentDetails[Context.INITIAL_CONTEXT_FACTORY] = parameters.initialContextFactory
+        environmentDetails["connectionfactory.factorylookup"] = parameters.factorylookup
+        environmentDetails["queue.sendQueue"] = parameters.sendQueue
+        environmentDetails["queue.receiveQueue"] = parameters.receiveQueue
+        return environmentDetails
+    }
+
     override fun start() {
         logger.info { "Starting the conn" }
-        val errorReporter : (Throwable) -> Unit = {}
-        val config : Map<String, String> = ObjectMapper().convertValue(parameters, object: TypeReference<Map<String, String>>(){})
+        val errorReporter : (Exception) -> Unit = {e -> reportError(e, {}) }
+        val config : Map<String, String> = toMap(parameters)
         client = AmqpClient(config, errorReporter)
 
-        client.start()
-
         val listener : (ByteArray) -> Unit = { bytes -> messageReceived(MessageHolder(bytes))}
         client.setMessageListener(listener)
     }
 
-    override fun send(message: Message) {
+    override fun send(message: RawMessage) {
         logger.debug { "Sending message: ${TextFormat.shortDebugString(message)}" }
         val bytes = message.toByteArray()
         client.send(bytes)
```

Comment:
comment for line:58 `messageSent(MessageHolder(bytes))`
passes property from the source RawMessage